jakarta.ls/src/main/resources/transactions.json

```diff
@@ -1,3 +1,53 @@
 {
-
+	"Initialize a UserTransaction object via Injection": {
+		"prefix": "userTransactionInjection",
+		"body": [
+			"@Resource",
+			"UserTransaction ${1:TransactionName};",
+			"",
+			"public void ${2:FunctionName}(){",
+			"\ttry {",
+			"\t\t${1:TransactionName}.begin();",
+			"",
+			"\t\t/* operation */",
+			"\t",
+			"\t\t${1:TransactionName}.commit();",
+			"\t} catch (Exception e1) {",
+			"\t\ttry {",
+			"\t\t\t${1:TransactionName}.rollback();",
+			"\t\t} catch (Exception e2){",
+			"\t\t\te2.printStackTrace();",
+			"\t\t}",
+			"\t}",
+			"}"
+		],
+		"description": "Initializes a UserTransaction object via Injection. Provides a template for begin(), commit(), and rollback() of the said UserTransaction object.",
+		"context": {
+			"type": "jakarta.transaction.UserTransaction"
+		}
+	},
+	"Initialize a UserTransaction object via JNDI Lookup": {
+		"prefix": "userTransactionJNDI",
+		"body": [
+			"UserTransaction ${1:TransactionName} = (UserTransaction) context.lookup(\"${2:Address}\");",
```

Comment:
As discussed we should not reference `context` without a corresponding variable - how is this typically used/initialized?

Reply:
You may be able to find an actual variable of type Context in the Java context where this snippet is used. If there isn't one you could use a default name but it won't compile. We could research how other snippets handle missing dependencies.

Reply:
I included a Context initialization in the most recent commit. Would it be a possible way to resolve this?